include/maths/CBasicStatistics.h

```diff
@@ -1327,10 +1330,18 @@ class MATHS_EXPORT CBasicStatistics {
 
     public:
         explicit COrderStatisticsHeap(std::size_t n, const LESS& less = LESS{})
-            : TImpl{std::vector<T>(n, T{}), less} {}
+            : TImpl{std::vector<T>(std::max(n, std::size_t(1)), T{}), less} {
+            if (n == 0) {
+                LOG_ERROR(<< "Invalid size of 0 for statistics accumulator");
```

Comment:
nit: I'd make this `Invalid size of 0 for order statistics accumulator`, since there are other types of statistics accumulators.